### PR TITLE
4710 administer dose modal

### PR DESCRIPTION
**Pull request by lache-melvin**
Part 1 of #4710 (Backend mutation to save the vaccination coming next)

# 👩🏻‍💻 What does this PR do?

Adds vaccination modal, capturing different input whether the dose was given or not given:

Screenshot 2024-09-06 at 9 52 52 AM

Screenshot 2024-09-06 at 9 52 37 AM

## 💌 Any notes for the reviewer?

# 🧪 Testing

You'll need to put a Vaccine course dose id in the `VaccinationsTab` component so the dose data is queried correctly

- [ ] Create an encounter for a patient enrolled in an immunisation program
- [ ] On vaccinations tab, open the modal (excuse the button haha)
- [ ] Dose name is populated in the header
- [ ] Encounter clinician is the default clinician
- [ ] Defaults to todays date - you can't select future dates
- [ ] Select `Given` - you can select an item (need to be configured in the vaccine course)
- [ ] Select `Not given` - you can select a reason

# 📃 Documentation

- [x] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: client/packages/system/src/Vaccination/Components/VaccinationModal.tsx [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+      />
+      <InputWithLabelRow
+        label={t('label.date')}
+        Input={
+          <DatePicker
+            disableFuture
+            value={draft.date}
+            onChange={date => updateDraft({ date })}
+            sx={{ flex: 1 }}
+          />
```
Comment (CarlosNZ) [inline client/packages/system/src/Vaccination/Components/VaccinationModal.tsx:141]: Question: do we plan to restrict the minimum date to the minimum age for the dose? Would it be hard to do now, or leave till later?

File: client/packages/system/src/Vaccination/api/useVaccination.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react';
+import { useQuery } from '@openmsupply-client/common';
+
+import { Clinician } from '../../Clinician';
+import { useVaccinationsGraphQL } from './useVaccinationsGraphQL';
+import { VACCINATION } from './keys';
+
+export interface VaccinationDraft {
```
Comment (CarlosNZ) [inline client/packages/system/src/Vaccination/api/useVaccination.ts:8]: Okay, good, you've started a new API group. I'll use this for the Card queries too.

**Review by CarlosNZ — APPROVED** [review]
Looks good thanks. Couple of small questions, nothing needed ASAP.

File: client/packages/system/src/Vaccination/Components/VaccinationModal.tsx (hunk 1 above)
Comment (lache-melvin) [reply client/packages/system/src/Vaccination/Components/VaccinationModal.tsx:141]: Thanks for spotting this! I checked with Mark - they'd like to leave it up to the clinician's discretion when they administer the dose 🙏

**Pull request merged**